README.markdown

```diff
@@ -187,49 +233,76 @@ If Puppet is running on a 64-bit system, the 32-bit Registry key can be explicit
+The title of the resource can determine the path and value name that will be managed, for example:
 
-If Puppet is running on a 64-bit system, the 32-bit Registry key can be explicitly managed using a prefix. For example: `32:HKLM\Software\Value3`.
+The manifest below would manage the value called 'foo' in the key 'HKLM\Software'
 
-##### `ensure`
+``` puppet
+registry_value { 'HKLM\Software\foo':
+  ...
+}
+```
 
-Tells Puppet whether the value should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: 'present'.
+You can also manage the default value of a key by adding a backslash at the end of the path.  The manifest below would manage the default value in the key 'HKLM\Software\Vendor'
 
-##### `type`
+``` puppet
+registry_value { 'HKLM\Software\Vendor\\':
+  ...
+}
+```
 
-*Optional.* Sets the data type of the specified value. Valid options: 'string', 'array', 'dword', 'qword', 'binary' and 'expand'. Default value: 'string'.
+Note that when explicitly setting a `path` or `value_name` parameter it is suggested to use both to avoid unexpected behavior, for example:
 
-##### `data`
+The manifest below would manage the value called 'value' (extracted from the title because value_name is not specified) in the key 'HKLM\Software\foo'
 
-Provides the contents of the specified value. Valid options: a string by default; an array if specified through the `type` parameter.
+``` puppet
+registry_value { 'title\value':
+  path => 'HKLM\Software\foo',
+  ...
+}
+```
 
-#### `registry::service`
+The manifest below would manage the default value (extracted from the title because value_name is not specified) in the key 'HKLM\Software\foo'
 
-##### `ensure`
+``` puppet
+registry_value { 'title':
+  path => 'HKLM\Software\foo',
+  ...
+}
+```
 
-Tells Puppet whether the service should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: 'present'.
 
-##### `display_name`
+**Parameters**
 
-*Optional.* Provides a Display Name for the service. Valid options: a string. Default value: the title of your declared resource.
+All parameters are required unless otherwise stated.
 
-##### `description`
+##### `path`
 
-*Optional.* Provides a description of the service. Valid options: a string. Default value: blank.
+*Optional.* Specifies a Registry value for Puppet to manage. Valid options: a string containing a Registry path. If any of the parent keys in the path don't exist, Puppet creates them automatically. For example: `HKLM\Software` or `HKEY_LOCAL_MACHINE\Software\Vendor`. Default value: the subkey in the title of the resource
 
-##### `command`
+If Puppet is running on a 64-bit system, the 32-bit Registry key can be explicitly managed using a prefix. For example: `32:HKLM\Software\Vendor`.
```

Comment:
Suggested rewording (active voice over passive):

"If Puppet is running on a 64-bit system, manage the 32-bit Registry key using a prefix".

Reply:
Fixed